src/framework/theme/components/select/select.component.ts

```diff
@@ -198,6 +198,11 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
    * */
   @Input() placeholder: string = '';
 
+  /**
+   * Custom comparing algorithm
+   */
+  @Input() compareWith: (o1: any, o2: any) => boolean = this.areIdentical;
```

Comment:
Please, specify type to prevent accidental call with arguments other than value type. Like this:
`(o1: T, o2: T) => boolean`.

Also, what do you think about adding setter to compareWith to prevent setting values other than functions?